src/mozanalysis/experiment.py

```diff
@@ -654,37 +696,67 @@ def _build_exposure_query(
 
     def _build_enrollments_query_normandy(
-        self, time_limits: TimeLimits, sample_size: int = 100
+        self,
+        time_limits: TimeLimits,
+        sample_size: int = 100,
     ) -> str:
         """Return SQL to query enrollments for a normandy experiment"""
-        return f"""
-        SELECT
-            e.client_id,
-            `mozfun.map.get_key`(e.event_map_values, 'branch')
-                AS branch,
-            MIN(e.submission_date) AS enrollment_date,
-            COUNT(e.submission_date) AS num_enrollment_events
-        FROM
-            `moz-fx-data-shared-prod.telemetry.events` e
-        WHERE
-            e.event_category = 'normandy'
-            AND e.event_method = 'enroll'
-            AND e.submission_date
-                BETWEEN '{time_limits.first_enrollment_date}' AND '{time_limits.last_enrollment_date}'
-            AND e.event_string_value = '{self.experiment_slug}'
-            AND e.sample_id < {sample_size}
-        GROUP BY e.client_id, branch
-            """  # noqa:E501
+        if self.experimental_unit == ExperimentalUnit.CLIENT:
+            return f"""
+            SELECT
+                e.client_id,
+                `mozfun.map.get_key`(e.event_map_values, 'branch')
+                    AS branch,
+                MIN(e.submission_date) AS enrollment_date,
+                COUNT(e.submission_date) AS num_enrollment_events
+            FROM
+                `moz-fx-data-shared-prod.telemetry.events` e
+            WHERE
+                e.event_category = 'normandy'
+                AND e.event_method = 'enroll'
+                AND e.submission_date
+                    BETWEEN '{time_limits.first_enrollment_date}' AND '{time_limits.last_enrollment_date}'
+                AND e.event_string_value = '{self.experiment_slug}'
+                AND e.sample_id < {sample_size}
+            GROUP BY e.client_id, branch
+                """  # noqa:E501
+        elif self.experimental_unit == ExperimentalUnit.GROUP:
+            # TODO: update this based on the final structure of the group_id
+            # within the events ping
+            return f"""
+            SELECT
+                e.profile_group_id,
+                `mozfun.map.get_key`(e.event_map_values, 'branch')
+                    AS branch,
+                MIN(e.submission_date) AS enrollment_date,
+                COUNT(e.submission_date) AS num_enrollment_events
+            FROM
+                `moz-fx-data-shared-prod.telemetry.events` e
+            WHERE
+                e.event_category = 'normandy'
+                AND e.event_method = 'enroll'
+                AND e.submission_date
+                    BETWEEN '{time_limits.first_enrollment_date}' AND '{time_limits.last_enrollment_date}'
+                AND e.event_string_value = '{self.experiment_slug}'
+                AND e.sample_id < {sample_size}
+            GROUP BY e.profile_group_id, branch
+                """  # noqa:E501
```

Comment:
Is it worth using a string var for client_id vs profile_group_id in an if statement outside the query so we don't have to have 2 copies of effectively the same query?

Reply:
Done, one version of the query now